cms/core/blocks/embeddable.py

```diff
@@ -98,3 +99,64 @@ def clean(self, value: "StructValue") -> "StructValue":
     class Meta:
         icon = "code"
         template = "templates/components/streamfield/ons_embed_block.html"
+
+
+class VideoEmbedBlock(blocks.StructBlock):
+    """A video embed block."""
+
+    embed_url = blocks.URLBlock(
+        help_text="""
+        The embed URL to the video hosted on YouTube or Vimeo, for example,
+        https://www.youtube.com/embed/{ video ID } or https://player.vimeo.com/video/{ video ID }
+        """
+    )
+    link_url = blocks.URLBlock(
```

Comment:
thought: the editorial experience feels subpar than just pasting a YouTube link (e.g. https://www.youtube.com/watch?v=Ffu-2jEdLPw) and things just working.

suggestion: extract the video id from the video URL and generate the embed URL from it.

Reply:
I did consider that. The reason I chose not to is because it would be different for youtube and vimeo (not a big deal), and because we're relying on a third party not to update the embed url pattern. However, happy to implement it if you think it's a good idea as I can see it would make life a lot easier for editors.

Reply:
This is now done. (I also updated the merge request description after my example vimeo link was removed)

Reply:
Just still fighting some linting issues.